docs/src/components/ComponentPage/IOSComponentPage.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow strict */
+
+import React from 'react';
+import BpkLink from 'bpk-component-link';
+
+import ComponentScreenshots from '../DocsPageBuilder/ComponentScreenshots';
+
+import ComponentPage from './ComponentPage';
+
+type Props = {
+  documentationId: string,
+  readme: string,
+  screenshots: [
+    {
+      id: string,
+      screenshots: {
+        altText: string,
+        height: number,
+        src: string,
+        subText: string,
+        title: string,
+        width: number,
+      },
+      title: string,
+    },
+  ],
+  usageTable: ?{
+    dos: Array<string>,
+    donts: Array<string>,
+  },
+};
+
+const IOSComponentPage = (props: Props) => {
+  const { documentationId, screenshots, readme, usageTable } = props;
+  const screenshotsAsExamples = screenshots.map(screenshot => ({
+    id: screenshot.id,
+    title: screenshot.title,
+    content: <ComponentScreenshots screenshots={screenshot.screenshots} />,
+  }));
+  return (
+    <ComponentPage
+      examples={screenshotsAsExamples}
+      readme={readme}
+      usageTable={usageTable}
+      additionalContent={[
+        {
+          id: 'docs',
+          title: 'Class reference',
+          content: (
+            <BpkLink
+              href={`/ios/versions/latest/Classes/${documentationId}.html`}
```

Comment:
Are we concerned that if the documentationId does not exist for some reason it would link to an empty page or not found?

Reply:
Flow should enforce this though, right?

Reply:
Right now I'm operating under the assumption that every iOS page has an accompanying docs page link.

I think it's preferable to keep using that assumption until it's proven wrong (we'd discover the error very quickly when we try to roll this out to all pages) than to add a null check that might not be needed.

Reply:
I guess but its better to be preventative and it not be an issue than, let it roll out and give our consumers a bad experience, then have to do a fix rather than have something in there that would prevent issues and broken functionality in the first place.

Plus we are already doing it for the Web page so it would kind of make sense to do it here?

Reply:
There's no chance of this affecting consumers, because if this does turn out to be a problem it'll immediately be caught during development when we try to use these new components on the remaining pages.

The web is a different case, because I have the opposite assumption there — that not all components have an accompanying sassdoc.

Reply:
Philosophically, I'd prefer to put something in the code that will cause a crash _during development_ so we can find and fix it, than put in a null check that'll fail quietly, which we might not notice.

This article (which is about Swift but the logic applies here) goes into more detail: https://wolfmcnally.com/82/force-unwrapping-swift-not-bad-thing

Reply:
@georgegillams yeah Flow will enforce it, which is why it can't affect any consumers because it'll be caught during development.

Reply:
Flow would capture it if the prop wasn't but what happens if we assume there is a doc id for a component but once its deployed, the link is generated based on the assumption but actually doesn't exist in the iOS/Android docs for whatever reason?

Flow can't actually test the url is valid and works but its just a small consideration to take into account here

Reply:
This is true, but I don't think that's relevant to this discussion as having a null check or not won't affect that.